### ISSUE TITLE
None.get error with aliasing the same name in CALL subqueries

### ISSUE THREAD

**Issue by tomocesnik** [neo4j/neo4j#12886]
While auto-generating Cypher code, I have found a strange error with using aliases to the same variable name in subqueries.

- Neo4j version: 4.4.4
- Operating system: Docker image "neo4j:4.4.4-community" on WSL2 and Docker Desktop on MS Windows 10 Pro
- API/Driver: Cypher in Neo4j Browser
- Steps to reproduce:
Pull the docker image and start the database instance.
Use the following Cypher code in Neo4j browser on empty database:
```
WITH 3 AS a
CALL {
  WITH a
  WITH a AS a
  CALL {
    WITH a
    RETURN a * a AS b
  }
  RETURN b
}
RETURN b
```
- Expected behavior
Query should succeed with 9 as a result.
- Actual behavior
Query returns an error with message:
```
Neo.DatabaseError.General.UnknownError
None.get
```
Log message in the console:
`ERROR Client triggered an unexpected error [Neo.DatabaseError.General.UnknownError]: None.get, reference 604af9ef-70f7-4258-b4a6-c0f844af36e1.`

The following queries work correctly and return the expected output.
Without subquery:
```
WITH 3 AS a
WITH a AS a
CALL {
  WITH a
  RETURN a * a AS b
}
RETURN b
```
Without aliasing in the subquery:
```
WITH 3 AS a
CALL {
  WITH a
  CALL {
    WITH a
    RETURN a * a AS b
  }
  RETURN b
}
RETURN b
```
With aliasing to different variable:
```
WITH 3 AS a
CALL {
  WITH a
  WITH a AS x
  CALL {
    WITH x
    RETURN x * x AS b
  }
  RETURN b
}
RETURN b
```
With aliasing to the same variable, but adding another alias to another variable:
```
WITH 3 AS a
CALL {
  WITH a
  WITH a AS a, 1 AS x
  CALL {
    WITH a
    RETURN a * a AS b
  }
  RETURN b
}
RETURN b
```

**Comment by HannesSandberg** [comment]
Thank you for your report! We will look into this and come back to you.

**Comment by arnefischereit** [comment]
Hi @tomocesnik,

It seems to me like the problematic query you mentioned works in 4.4.6. Can you confirm?

**Comment by tomocesnik** [comment]
You are correct, this seems to be fixed on Neo4j 4.4.6 version. I guess you can resolve this issue?
